### PR TITLE
Add Contested Proposals filtering

### PR DESCRIPTION
**Pull request by JSKitty**
## Abstract

To help "dust away" and disincentivise spam or malicious proposals, this PR ensures proposals with less than 25% net consensus (i.e; 75% downvotes) are considered Contested Proposals.

The only exception is proposals with less than 100 votes, to prevent "downvote bombing" on young proposals.

Contested Proposals are effectively "punished" by:
- Being dropped to the lowest part of the screen in a "Contested Proposals" section.
- Being given a slightly lower opacity, to move the users' eyes to higher proposals.

This is purely a display change, there's no change in voting functionality, the user still has full and total freedom to vote on proposals they wish; the PR does not intend to harm the average proposal, but act as a 'net' for proposals that are **most likely** to be spam or otherwise malicious.

image

As usual, related code was also cleaned up and simplified slightly, as well as given JSDoc coverage and translation templates.

## What does this PR address?
It addresses possible spam or ill-intent proposals by lowering their presence in the Governance menu - the PR intends to balance "Freedom" and "Censoring" in a way that clearly discourages spam proposals, without impairing the freedom of users.

## What features or improvements were added?
A "Contested Proposals" table to the Governance menu, which displays exclusively proposals with a net of **25% or less Net Upvotes** or equally **75% Downvotes**.

## How does this benefit users?
This should help guide users' attention away from clearly-malicious proposals, such as spam, which clears up the primary Governance Proposals section for well-vetted proposals by the DAO.

### CODE REVIEW

File: scripts/global.js [hunk 1]
```diff
@@ -921,52 +923,81 @@ export async function restoreWallet() {
...
-        td2.innerHTML = `<b>${sanitizeHTML(proposal.MonthlyPayment)}</b> ${
+    // Fetch all proposals from the network
+    const arrProposals = await Masternode.getProposals();
+
+    /* Sort proposals into two categories
+        - Standard
+        - Contested (When a proposal may be considered spam, malicious, or simply highly contestable)
+    */
+    const arrStandard = arrProposals.filter(a => a.RemainingPaymentCount > 0 && a.Ratio > 0.25);
+    const arrContested = arrProposals.filter(a => a.RemainingPaymentCount > 0 && a.Ratio <= 0.25);
```
Comment (panleone) [inline scripts/global.js:938]: Without enough statistics it does not make any sense to call a proposal contested.  I'd add the condition that the total number of votes is at least 100 (or any big enough number)

**Review by panleone — CHANGES_REQUESTED** [review]
Good idea, only one small comment

File: scripts/global.js (hunk 1 above)
Comment (shrnkld) [reply scripts/global.js:938]: I'd also suggest renaming the section to "Other Proposals" as _contested_ is subjective and will possibly not be true for all conditions.

Comment (JSKitty) [reply scripts/global.js:938]: Good catch on a 'minimum' of votes, panleone - adding this! :+1:
I think 100 votes minimum is fine.

@shrnkld The "subtext" for the category explains that it's not a "deterministic" decision, just a kind of catch-all net for anything that may be **likely** spam or 'contestable'.

I also would rather not go against what the community voted for; as the choice was given to them.

File: scripts/masternode.js [hunk 2]
```diff
@@ -271,11 +271,20 @@ export default class Masternode {
...
 
     /**
-     * @return {Promise<Array>} A list of currently active proposal
+     * 
+     * @param {object} options
+     * @param {bool} options.fAllowFinished - Pass `true` to stop filtering proposals if finished
+     * @return {Promise<Array<object>} A list of currently active proposal
      */
-    static async getProposals() {
+    static async getProposals({ fAllowFinished = false }) {
```
Comment (Duddino) [inline scripts/masternode.js:279]:
```suggestion
    static async getProposals({ fAllowFinished = false } = {}) {
```
Add default parameter value

**Review by Duddino — COMMENTED** [review]
Last nitpick as far as i'm concerned

**Review by Duddino — APPROVED** [review]
tACK, code is clean and works

**Review by panleone — APPROVED** [review]
tACK

**Pull request merged**